### PR TITLE
Update CSINodeInfo

### PR DESCRIPTION
**Pull request by jsafrane**
Kubernetes 1.13 introduced `CSINodeInfo.Spec` + `Status`. Fix the code and vendor kubernetes 1.13.0-beta2.

Fixes: #99

@xing-yang, I fixed your PR to vendor the right code from the right places (I think :-).

### CODE REVIEW

**Comment by xing-yang** [comment]
Thanks @jsafrane !  I'll give this PR a try.

**Comment by xing-yang** [comment]
I tested this PR with and without enabling CSINodeInfo, but I still got the error:

external-attacher logs:

I1126 20:03:33.457293       1 csi_handler.go:524] Can't get CSINodeInfo 127.0.0.1: csinodeinfos.csi.storage.k8s.io "127.0.0.1" not found
I1126 20:03:33.457309       1 csi_handler.go:388] Saving attach error to "csi-cc697e353e2d1b67f03d4c6e37d7f7c04480c3126d12b036e1e29c3517bd0512"
I1126 20:03:33.459175       1 csi_handler.go:398] Saved attach error to "csi-cc697e353e2d1b67f03d4c6e37d7f7c04480c3126d12b036e1e29c3517bd0512"
I1126 20:03:33.459194       1 csi_handler.go:103] Error processing "csi-cc697e353e2d1b67f03d4c6e37d7f7c04480c3126d12b036e1e29c3517bd0512": failed to attach: node "127.0.0.1" has no NodeID annotation

driver-registrar logs:

I1126 20:00:30.487440       1 connection.go:136] GRPC call: /csi.v1.Node/NodeGetInfo
I1126 20:00:30.487454       1 connection.go:137] GRPC request: 
I1126 20:00:30.495795       1 connection.go:139] GRPC response: node_id:"ubuntu,iqn:iqn.1993-08.org.debian:01:c6961059376e" 
I1126 20:00:30.496273       1 connection.go:140] GRPC error: <nil>
I1126 20:00:30.496526       1 node_register.go:63] CSI driver node ID: "ubuntu,iqn:iqn.1993-08.org.debian:01:c6961059376e"
I1126 20:00:30.497795       1 node_register.go:86] Starting Registration Server at: /registration/csi-opensdsplugin-reg.sock
I1126 20:00:30.499127       1 node_register.go:93] Registration Server started at: /registration/csi-opensdsplugin-reg.sock

**Comment by davidz627** [comment]
@xing-yang are you using the new plugin registration dir?
https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/blob/master/deploy/kubernetes/base/node.yaml#L64
`path: /var/lib/kubelet/plugins_registry/`
as opposed to the old: `path: /var/lib/kubelet/plugins`

**Comment by xing-yang** [comment]
Yes.  Actually I realized I have been testing using K8S v1.13.0-beta.1 so my tests are invalid.  I need to re-test using beta.2.

**Comment by xing-yang** [comment]
Tested with CSINodeInfo enabled using K8S v1.13.0-beta.2.  It works!
Will test without CSINodeInfo next.

**Comment by xing-yang** [comment]
Tested without CSINodeInfo using K8S v1.13.0-beta.2.  It works as well!

**Comment by xing-yang** [comment]
LGTM

File: Gopkg.toml [hunk 1]
```diff
@@ -20,18 +20,24 @@
   name = "google.golang.org/grpc"
   version = "1.9.2"
 
-[[constraint]]
+[[override]]
+  name = "github.com/json-iterator/go"
+  version = "1.1.4"
+
+// TODO: use [[constraint]] when kubernetes-csi/kubernetes-csi-migration-library uses 1.13 code
```
Comment (davidz627) [inline Gopkg.toml]: https://github.com/kubernetes-csi/kubernetes-csi-migration-library/pull/9

**Comment by saad-ali** [comment]
Let's cherry pick this to `release-1.0` branch

**Comment by jsafrane** [comment]
Picked in #103

**Pull request merged**